app/src/models/status.ts

```diff
@@ -2,32 +2,99 @@ import { DiffSelection } from './diff'
 import { OcticonSymbol } from '../ui/octicons'
 import { assertNever } from '../lib/fatal-error'
 
-/** the state of the changed file in the working directory */
-export enum FileStatus {
+/**
+ * The status entry code as reported by Git.
+ *
+ * NOTE: 'U' is considered "Updated but unmerged", and is a valid code here,
+ * but instead we mark this as "Modified" when surfacing conflicts.
+ *
+ * TODO (BF): what as I thinking here? Tidy up this comment or remove it.
+ */
+export enum GitStatusEntry {
+  // M
+  Modified,
+  // A
+  Added,
+  // D
+  Deleted,
+  // R
+  Renamed,
+  // C
+  Copied,
+  // .
+  Unchanged,
+  // ?
+  Untracked,
+  // !
+  Ignored,
+}
+
+/** The file status as represented in GitHub Desktop. */
+export enum AppFileStatus {
   New,
   Modified,
   Deleted,
+  Copied,
   Renamed,
   Conflicted,
-  Copied,
 }
 
+/** The porcelain status for an ordinary changed entry */
+type OrdinaryEntry = {
+  readonly kind: 'ordinary'
+   /** how we should represent the file in the application */
+  readonly type: 'added' | 'modified' | 'deleted',
+  /** the staged status of the file (if known) */
+  readonly staged?: GitStatusEntry,
```

Comment:
`staged` feels like a funny name to me. I tend to think of "staged" as a boolean indicating whether the entry exists in the index.

The `git status` man page seems to use phrases like "status of the index" vs. "status of the work tree." Since this PR is trying to conform more to git, I think our phrasing should too. Thoughts?